error-handling/src/main/java/com/palantir/remoting1/errors/SerializableError.java

```diff
@@ -37,36 +36,30 @@
 
     public abstract String getMessage();
 
+    // TODO(rfink) This needs to get renamed once a wire-break is acceptable.
     @JsonProperty("exceptionClass")
-    public abstract String getExceptionClassName();
-
-    /**
-     * @deprecated Use {@link #getExceptionClassName()} instead.
-     */
-    @Deprecated
-    @JsonIgnore
-    @SuppressWarnings("unchecked")
-    public final Class<? extends Exception> getExceptionClass() {
-        try {
-            return (Class<? extends Exception>) Class.forName(getExceptionClassName());
-        } catch (ClassNotFoundException e) {
-            throw new RuntimeException(e);
-        }
-    }
+    public abstract String getExceptionName();
 
     @Nullable
     public abstract List<StackTraceElement> getStackTrace();
 
+    /** Constructs a new error whose exception name is the fully-qualified name of the given class. */
     public static SerializableError of(String message, Class<? extends Exception> exceptionClass) {
```

Comment:
No strong conviction here; at the moment all error names are fully
qualified exception names by construction. You're effectively suggesting
that exception producers declare something like
`MyServiceErrorTypes.FOO_BAR_ERROR` and then use those error when
constructing exceptions? Most SerializedErrors will likely be produced by
the jerey exception mappers, so it's really no difference for users unless
they want to construct their own, custom exceptions for some reason.

On Tue, Aug 16, 2016 at 4:38 AM Mark Elliot <notifications@github.com>
wrote:

> In
> error-handling/src/main/java/com/palantir/remoting1/errors/SerializableError.java
> <https://github.com/palantir/http-remoting/pull/199#discussion_r74918404>:
>
> >
> >      @Nullable
> >      public abstract List<StackTraceElement> getStackTrace();
> >
> > +    /** Constructs a new error whose exception name is the fully-qualified name of the given class. */
> >      public static SerializableError of(String message, Class<? extends Exception> exceptionClass) {
>
> Should we change the class parameter to be a string?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/http-remoting/pull/199/files/d620e6e9534492bdaf02de0938714cb37e370b24#r74918404>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGOdwfJXU5IYrRgoCQ2EeAxCVyB5Fzryks5qgaEmgaJpZM4Jjywx>
> .
>